packages/react-i18n/src/index.tsx

```diff
@@ -66,9 +67,28 @@ export const useI18n = (): I18nProps => {
  * }
  * export default withI18n( MyComponent );
  */
-export const withI18n = createHigherOrderComponent< I18nProps >( InnerComponent => {
+export const withI18n = createHigherOrderComponent< I18nReact >( InnerComponent => {
 	return props => {
 		const i18n = useI18n();
 		return <InnerComponent { ...i18n } { ...props } />;
 	};
 }, 'withI18n' );
+
+/**
+ * Utility to make a new context value
+ *
+ * @param locale The locale of the context value
+ * @param localeData The localeData
+ *
+ * @returns The context value with bound translation functions
+ */
+function makeContextValue( locale: string, localeData?: LocaleData ): I18nReact {
+	const i18n = new I18n( localeData );
+	return {
```

Comment:
What are your thoughts with this abstraction of the `i18n` instance?

Reply:
Yes, this combined with the memoization is intended to trigger renders when necessary otherwise not.

In previous work I added `i18nLocale` to trigger rerenders, but the `bind`s here should be sufficient to trigger rerenders. I'll remove it.

The other bit here is exposing the desired parts of the API. This exposes the translate functions, but none of the lower level internals or funcitonality to modify locale data in the instance.